File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        
        Stack<String> stack = new Stack<>();
        stack.push("base");
        for(int i=0; i<s.length(); i++){
            if(s.charAt(i) == '(')
                stack.push("(");
            else if ( s.charAt(i) == ')' && stack.peek().equals("(") )
                stack.pop();
            
            else{
                answer = false;
                break;
            }
        }
        
        
        if(answer && stack.peek().equals("("))
            return false;
        return answer;
    }
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    for i in range(len(s)):
        if s[i] == "(":
            stack.append(s[i])
        elif s[i] == ")":
            if not stack:
                return False
            if stack[-1] == ")":
                return False
            else:
                del stack[-1]
    if stack:
        return False
    return True